### PR TITLE
Upgrade terraform-provider-azurerm to v3.111.0

### LINKED ISSUE #2201: Upgrade terraform-provider-azurerm to v3.111.0
Release details: https://github.com/hashicorp/terraform-provider-azurerm/releases/tag/v3.111.0

### PR DESCRIPTION
**Pull request by danielrbradley**
This PR was generated via `$ upgrade-provider pulumi/pulumi-azure --upstream-provider-name=terraform-provider-azurerm --kind=provider`.

---

- Upgrading terraform-provider-azurerm from 3.110.0  to 3.111.0.
	Fixes #2201

### CODE REVIEW

File: patches/0009-Make-fields-case-insensitive-for-backward-compatibil.patch [hunk 1]
```diff
@@ -323,7 +323,7 @@ index 5d1d4cfe1c..72cb85465c 100644
  
  			"dns_name_label": {
 diff --git a/internal/services/containers/kubernetes_cluster_resource.go b/internal/services/containers/kubernetes_cluster_resource.go
-index e40ce7505b..337dfef428 100644
+index 6917db6163..bcbc450370 100644
```
Comment (VenelinMartinov) [inline patches/0009-Make-fields-case-insensitive-for-backward-compatibil.patch:326]: are the patch diffs meaningful?

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+  "go.testTags": "all",
```
Comment (VenelinMartinov) [inline .vscode/settings.json:2]: accidental vscode directory commit?

**Review by VenelinMartinov — APPROVED** [review]
Looks good!

File: .vscode/settings.json (hunk 2 above)
Comment (danielrbradley) [reply .vscode/settings.json:2]: Bah yeah, but actually helpful to have there ... think this is caused by the upgrade tool committing everything

**Pull request merged**